.github/workflows/test.yml

```diff
@@ -0,0 +1,46 @@
+name: "Test - action"
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  typecheck:
```

Comment:
I'd make only one job with both typecheck and test, as most of the job time is spent initializing